### ISSUE TITLE
Cannot use destructured tuple fields as keys in maps

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#10074]
**Description:**
The following program results in an index out of bounds exception internally. 
```ballerina
function main(string... args) {
        map m;
        (string, any) tup = ("test", 5.4);
        var (field, value) = tup;
        m[field] = "Test";
}
```
The above program gives the following output:
```
error: error, message: 7
	at main(test.bal:3)
```
The following doesn't work either.
```ballerina
function main(string... args) {
        map m;
        (string, any) tup = ("test", 5.4);

        string field;
        any value;
        (field, value) = tup;
        m[field] = "Test";
}
```

When this is fixed, need to enable the two disabled test cases in the record iterations tests.

**Affected Versions:**
0.981.0

**Comment by Anoukh** [comment]
@pubudu91 this might be fixed with PR #9752

**Comment by SupunS** [comment]
Fixed in v0.981.1-SNAPSHOT
